openhands/runtime/plugins/agent_skills/file_ops/file_ops.py

```diff
@@ -511,94 +511,96 @@ def _edit_file_impl(
         # because the env var will be set AFTER the agentskills is imported
         if enable_auto_lint:
             # BACKUP the original file
-            original_file_backup_path = os.path.join(
-                os.path.dirname(file_name),
-                f'.backup.{os.path.basename(file_name)}',
-            )
-            with open(original_file_backup_path, 'w') as f:
-                f.writelines(lines)
-
-            lint_error, first_error_line = _lint_file(file_name)
-
-            # Select the errors caused by the modification
-            def extract_last_part(line):
-                parts = line.split(':')
-                if len(parts) > 1:
-                    return parts[-1].strip()
-                return line.strip()
-
-            def subtract_strings(str1, str2) -> str:
-                lines1 = str1.splitlines()
-                lines2 = str2.splitlines()
-
-                last_parts1 = [extract_last_part(line) for line in lines1]
-
-                remaining_lines = [
-                    line
-                    for line in lines2
-                    if extract_last_part(line) not in last_parts1
-                ]
-
-                result = '\n'.join(remaining_lines)
-                return result
-
-            if original_lint_error and lint_error:
-                lint_error = subtract_strings(original_lint_error, lint_error)
-                if lint_error == '':
-                    lint_error = None
-                    first_error_line = None
-
-            if lint_error is not None:
-                if first_error_line is not None:
-                    show_line = int(first_error_line)
-                elif is_append:
-                    # original end-of-file
-                    show_line = len(lines)
-                # insert OR edit WILL provide meaningful line numbers
-                elif start is not None and end is not None:
-                    show_line = int((start + end) / 2)
-                else:
-                    raise ValueError('Invalid state. This should never happen.')
-
-                ret_str += LINTER_ERROR_MSG
-                ret_str += lint_error + '\n'
-
-                editor_lines = n_added_lines + 20
-                sep = '-' * 49 + '\n'
-                ret_str += (
-                    f'[This is how your edit would have looked if applied]\n{sep}'
-                )
-                ret_str += (
-                    _print_window(file_name, show_line, editor_lines, return_str=True)
-                    + '\n'
+            with tempfile.TemporaryDirectory() as temp_dir:
+                original_file_backup_path = os.path.join(
+                    temp_dir,
+                    f'.backup.{os.path.basename(file_name)}',
```

Comment:
I'd put double-quotes around it to prevent issues with spaces in filenames

Reply:
spaces...

Reply:
How about using temp file directly? (if it passes test ofc)
https://github.com/All-Hands-AI/OpenHands/pull/3958/commits/deed9d6e2f65dac47e32826284bac00167962a09

Reply:
That should work as the generated filename should be quite unique.